src/main/java/core/basesyntax/dao/ma/PersonDaoImpl.java

```diff
@@ -11,6 +13,24 @@ public PersonDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Person save(Person person) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(person);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't add person: "
+                    + person + " to the DB. ", e);
```

Comment:
We print superclasses in exceptions messages, so let's override **toString** methods in them.